File: src/wheel/config/config_factory.py
from src.common import Factory
from .config_loader import ConfigLoader
from .imp.yaml_config_loader import YamlConfigLoader


class ConfigLoaderFactory(Factory[ConfigLoader]):
    def create(self, config: object) -> ConfigLoader:
        if isinstance(config, str):
            return YamlConfigLoader(config)
        return YamlConfigLoader()

File: lynne-cpp/src/wheel/config/config_factory.cpp
#include "wheel/config/config_factory.h"
#include "wheel/config/imp/json_config_loader.h"

namespace lynne {
namespace wheel {

ConfigLoader* ConfigLoaderFactory::create(const char* path) const {
    if (path != nullptr && path[0] != '\0') {
        return new JsonConfigLoader(std::string(path));
    }
    return new JsonConfigLoader("config.json");
}

} // namespace wheel
} // namespace lynne
